providers/vertica/tests/unit/vertica/hooks/test_vertica_sql.py

```diff
@@ -0,0 +1,272 @@
+        pytest.param(
+            False,
+            False,
+            "SELECT * FROM table",
+            ["SELECT * FROM table"],
+            [("1", "row1"), ("2", "row2")],
+            [SerializableRow("1", "row1"), SerializableRow("2", "row2")],
+            id="Single query, return_last=False",
+        ),
+        pytest.param(
+            True,
+            True,
+            "SELECT * FROM table1; SELECT * FROM table2;",
+            ["SELECT * FROM table1;", "SELECT * FROM table2;"],
+            [[("1", "row1"), ("2", "row2")], [("3", "row3"), ("4", "row4")]],
+            [SerializableRow("3", "row3"), SerializableRow("4", "row4")],
+            id="Multiple queries, split_statements=True, return_last=True",
+        ),
+        pytest.param(
+            True,
+            False,
+            "SELECT * FROM table1; SELECT * FROM table2;",
+            ["SELECT * FROM table1; SELECT * FROM table2;"],
+            [("1", "row1"), ("2", "row2")],
+            [SerializableRow("1", "row1"), SerializableRow("2", "row2")],
+            id="Multiple queries, split_statements=False",
+        ),
+        pytest.param(
+            True,
+            False,
+            "SELECT * FROM empty",
+            ["SELECT * FROM empty"],
+            [],
+            [],
+            id="Empty result",
+        ),
+    ],
+)
+def test_vertica_run_queries(
+    vertica_hook,
+    mock_cursor,
+    return_last,
+    split_statements,
+    sql,
+    expected_calls,
+    cursor_results,
+    expected_result,
+):
+    if split_statements:
+        mock_cursor.fetchall.side_effect = cursor_results
+        mock_cursor.nextset.side_effect = [True] * (len(cursor_results) - 1) + [False]
+    else:
+        mock_cursor.fetchall.return_value = cursor_results
+        mock_cursor.nextset.side_effect = lambda: False
+
+    result = vertica_hook.run(
+        sql,
+        handler=lambda cur: cur.fetchall(),
+        split_statements=split_statements,
+        return_last=return_last,
+    )
+
+    actual_calls = [call.args[0] for call in mock_cursor.execute.call_args_list]
+    assert actual_calls == expected_calls
+    assert [SerializableRow(*row) for row in result] == expected_result
+
+
+def test_run_with_multiple_statements(vertica_hook, mock_cursor):
+    mock_cursor.fetchall.side_effect = [[(1,)], [(2,)]]
+
+    mock_cursor.nextset.side_effect = [True, False]
+
+    sql = "SELECT 1; SELECT 2;"
+
+    results = vertica_hook.run(sql, handler=lambda cur: cur.fetchall(), split_statements=True)
+
+    mock_cursor.execute.assert_has_calls(
+        [
+            call("SELECT 1;"),
+            call("SELECT 2;"),
+        ]
+    )
+
+    assert results == [(2,)]
+
+
+def test_get_uri(vertica_hook):
+    """
+    Test that the get_uri() method returns the correct connection string.
+    """
+
+    expected_uri = "vertica-python://user:pass@vertica.cloud.com:5433/test_db"
+
+    actual_uri = vertica_hook.get_uri()
+
+    assert actual_uri == expected_uri
```

Comment:
```suggestion
    assert vertica_hook.get_uri() == "vertica-python://user:pass@vertica.cloud.com:5433/test_db"
```

 i think it's easier to read this way